buildlib/azure-pipelines.yml

```diff
@@ -38,4 +38,10 @@ stages:
           - publish: '$(System.DefaultWorkingDirectory)/'
             artifact: jar for spark-$(spark_version)
             displayName: Publish jars
-
+      - job: test
```

Comment:
Each job runs in its own env and directory, so you won't have a locally built jar file like that, you will need to rebuild. In addition, you might want to run for all Spark versions. In this case, you can make this job a step under the `Build` stage and define `$SPARK_UCX_JAR` depends on Spark version.

Reply:
Is it possible to run step with 1 matrix parameter? In the previous step we building jars for 2 spark versions, but we can test just with spark-2.4

Reply:
No agent found in pool MLNX which satisfies the specified demands:
 Maven
 Agent.Version -gtVersion 2.160.1

Reply:
> Is it possible to run step with 1 matrix parameter? In the previous step we building jars for 2 spark versions, but we can test just with spark-2.4

No, matrix applied to a job. If you don't need to test all version, your approach is good. But you need to pull jar file as an artifact, see [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml#downloading-artifacts).

Reply:
> No agent found in pool MLNX which satisfies the specified demands:
> Maven
> Agent.Version -gtVersion 2.160.1

Our agents don't report `Maven` capability, I need to check how to do it. Anyway, currently only one host is available and it doesn't have any MLNX devices anyway.

Reply:
Spark-2.3 is not supported, so for now we can build and test only for spark-2.4